File: internal/server/models/agent_identity.go
package models

import "time"

// AgentIdentity stores AI Agent identity records for non-K8s deployments.
// Each agent gets an independent identity with tool RBAC and sandbox settings.
type AgentIdentity struct {
	Model

	TenantID          string     `gorm:"size:36;index;not null" json:"tenant_id"`
	Name              string     `gorm:"size:200;not null" json:"name"`
	PeerRef           string     `gorm:"size:200;not null" json:"peer_ref"`
	AllowedTools      string     `gorm:"type:text" json:"allowed_tools,omitempty"`      // JSON array
	AllowedNamespaces string     `gorm:"type:text" json:"allowed_namespaces,omitempty"` // JSON array
	Sandbox           string     `gorm:"size:50;not null;default:none" json:"sandbox"`
	AuditLevel        string     `gorm:"size:50;not null;default:write" json:"audit_level"`
	EnforcementMode   string     `gorm:"size:50;not null;default:enforce" json:"enforcement_mode"`
	ExpiresAt         *time.Time `json:"expires_at,omitempty"`
	ParentRef         string     `gorm:"size:200" json:"parent_ref,omitempty"`
	SpawnableRoles    string     `gorm:"type:text" json:"spawnable_roles,omitempty"` // JSON array
	Phase             string     `gorm:"size:50;not null;default:Pending" json:"phase"`
	PeerIP            string     `gorm:"size:64" json:"peer_ip,omitempty"`
	LastSeenAt        *time.Time `json:"last_seen_at,omitempty"`
	Description       string     `gorm:"size:500" json:"description,omitempty"`
}

func (AgentIdentity) TableName() string { return "t_agent_identity" }

File: frontend/src/api/agent-identity.ts
import request from '@/api/request'

function wsNS(): string {
  return localStorage.getItem('active_ws_id') || ''
}

// ── Types (matches backend model) ──────────────────────────────

export interface AgentIdentity {
  id: string
  tenant_id: string
  name: string
  peer_ref: string
  allowed_tools?: string // JSON array
  allowed_namespaces?: string // JSON array
  sandbox: string
  audit_level: string
  enforcement_mode: string
  expires_at?: string
  parent_ref?: string
  spawnable_roles?: string // JSON array
  phase: string
  peer_ip?: string
  last_seen_at?: string
  description?: string
  created_at?: string
  updated_at?: string
}

export interface AgentIdentityInput {
  name: string
  peer_ref: string
  allowed_tools?: string[]
  allowed_namespaces?: string[]
  sandbox?: string
  audit_level?: string
  enforcement_mode?: string
  parent_ref?: string
  spawnable_roles?: string[]
  description?: string
}

// ── API functions ────────────────────────────────────────────────

export const listAgentIdentities = (): Promise<{ data: AgentIdentity[] }> =>
  request.get('/agent-identities', { namespace: wsNS() })

export const getAgentIdentity = (id: string): Promise<{ data: AgentIdentity }> =>
  request.get(`/agent-identities/${id}`)

export const createAgentIdentity = (input: AgentIdentityInput): Promise<{ data: AgentIdentity }> =>
  request.post(`/agent-identities?namespace=${encodeURIComponent(wsNS())}`, input)

export const updateAgentIdentity = (id: string, input: AgentIdentityInput): Promise<{ data: AgentIdentity }> =>
  request.put(`/agent-identities/${id}`, input)

export const deleteAgentIdentity = (id: string): Promise<void> =>
  request.delete(`/agent-identities/${id}`)
